app/controllers/public/sessions.js

```diff
@@ -17,24 +17,26 @@ export default class SessionsController extends Controller {
   @computed('model.session.@each', 'timezone')
   get groupByDateSessions() {
     let sessions;
-
     if (this.sort === 'title') {
       sessions = groupBy(this.model.session.toArray(), '');
     } else {
       sessions = groupBy(this.model.session.toArray(), s => moment.tz(s.startsAt, this.timezone).format('dddd, Do MMMM'));
     }
-
     return Object.entries(sessions).map(([date, sessions]) => ({ date: date === 'undefined' ? null : date, sessions }));
   }
 
-  @computed('model.event.startsAt', 'model.event.endsAt', 'timezone')
+  @computed('model.dates', 'timezone')
   get allDates() {
-    const arr = [];
-    const difference = (this.model.event.endsAt).diff(this.model.event.startsAt, 'days');
-    for (let i = 0; i <= Math.abs(difference); i++) {
-      arr.push(moment.tz(this.model.event.startsAt, this.timezone).add(i, 'days').toISOString());
-    }
-    return arr;
+
+    const arr = this.model.dates.toArray().map(el => moment.tz(el.startsAt, this.timezone).toISOString());
+
+    const uniqueDates = new Set();
+
+    arr.sort().forEach(el => {
+      uniqueDates.add(moment(el).format('YYYY-MM-DD'));
```

Comment:
This can be moved to .sort().map above

Reply:
map().sort() will do the correct job sir

Reply:
This can be moved to .map above